src/Razor/src/Microsoft.AspNetCore.Razor.Language/Components/ComponentBindLoweringPass.cs

```diff
@@ -171,8 +171,10 @@ private IntermediateNode[] RewriteUsage(IntermediateNode parent, TagHelperProper
                 node.AttributeName,
                 out var valueAttributeName,
                 out var changeAttributeName,
+                out var expressionAttributeName,
                 out var valueAttribute,
-                out var changeAttribute))
+                out var changeAttribute,
+                out var expressionAttribute))
```

Comment:
I'd really like to see the comments above this method updated with some of these new details (similar to what you put in the PR description).